### ISSUE TITLE
Forward Followers to my main account. 

### ISSUE THREAD

**Issue by pkreissel** [Automattic/wordpress-activitypub#834]
### What

I would like anyone who tries to follow my blog to be redirected to my main profile instead. 

### Why

Currently we don't want to use the mastodon profile for Wordpress, because we already have a big identity on Mastodon [here](https://digitalcourage.social/@Volksverpetzer). We also post other stuff not just our articles. But it would be awesome to use the comments from this plugin. 

But if we use this plugin and then reblog its posts, the problem would be, that anyone who sees these posts, would follow the small Wordpress profile, and not the big main profile that also posts other stuff. 

That's why it would be great if they could be redirected to the main profile. 

### How

I think the [MOVE activity](https://swicg.github.io/activitypub-data-portability/#move-action) would already tick a lot of the boxes with regard to followers. However right now it kind of assumes the old identity to shut down, but does not really specify that. 

In Wordpress the plugin could specify a "main profile" that this Wordpress activity pub actor would redirect to using  the specification from the MOVE activity.

**Comment by pfefferle** [comment]
This is not possible with ActivityPub (the spec) at the moment! As I already commented on the linked W3C issue, `Move` means to move all the followers, so they will no longer follow your blogs account and will not receive any updates any more.

The only possible solution would be, if you boost the blog-posts instead of cross-posting them. This way, the followers of your Mastodon account would receive them and they wouldn't have to follow your blogs account(s) to comment/like/boost.

What you have to do: Follow your blog from your Mastodon Account -> Publish a post -> Boost that post when you see it in your Mastodon timeline.

I am experimenting with an automatism, but it is not yet working properly: https://github.com/pfefferle/wordpress-crossbooster

**Comment by pfefferle** [comment]
Btw. We are actively working on the Move feature atm. https://github.com/Automattic/wordpress-activitypub/pull/685

**Comment by pfefferle** [comment]
There is also a way, to do it by hand without an automatism. You could copy/paste the URL of a blog-post into the search field of mastodon, mastodon should index it then so that you can boost it. At least in theory...

**Comment by pfefferle** [comment]
Oh, you just had a look at your example link, I feel a bit honored that you used the plugin for the Volksverpetzer blog (I am a big fan) and that makes me even more sad that you want to `Move` all the followers 😉

**Comment by pkreissel** [comment]
Thank you for being patient with my crazy requests <3

If I just copy&paste the link and then reblog with the main account, then people who see reblogs of that post by our followers will follow the blog, but not the main account. Then the followship of the blog would be split among accounts, which is neither ideal for us, nor for our followers.

I could also imagine that we use the Wordpress Plugin as our main account (can the plugin / our server handle 100 000 subscribers 🫣), but then we would still need to MOVE from the main account to the plugin AND the plugin would need to allow for posts and threads that dont appear on the blog. 

So right now this would be perfect:
- The Wordpress blog is not followed by anyone
- It has a Move forwarding enabled, so that anyone who still looks at the profile somehow will immediately be forwarded to the main account, like in the example profile of myself on chaos.social.
- I post an article on the blog -> this generates an activity pub object.
- Now I can either manually or via web hook endpoint reblog that activity pub object with the main account (e.g. by searching for it manually or via API) 
- Now all of our followers will receive the object (I hope)

I think this will be the result (correct me if wrong)
- If they click on the profile of the object they get redirected to the main account. 
- If they comment on the object the comments show up on the blog. 
- Most of the work is done by the Mastodon Server of the main account and not the web server of the blog, that is already overworked as it is. 

I think there would still be some bumps along the way, but most of that would be for the Mastodon-side to solve.

**Comment by pfefferle** [comment]
> If I just copy&paste the link and then reblog with the main account, then people who see reblogs of that post by our followers will follow the blog, but not the main account. Then the followship of the blog would be split among accounts, which is neither ideal for us, nor for our followers.

No, I think about solution to your "after the Move" problem! As I said, part of the move is, that all accounts unfollow your blog and the ability to follow will be blocked. So how to get your posts to the fediverse. And one solution would be to copy&paste them to the search bar.

To your points, yes this should work!

> I could also imagine that we use the Wordpress Plugin as our main account (can the plugin / our server handle 100 000 subscribers 🫣), but then we would still need to MOVE from the main account to the plugin AND the plugin would need to allow for posts and threads that dont appear on the blog.

I am not sure if that is a question about if the plugin can handle it, but if the Webserver can handle it. We definitely have to optimize the distribution a bit, because we can't send 100.000 posts in one request, but the queries from the followers and their followers might be hard to handle by the Webserver: https://ar.al/2022/11/09/is-the-fediverse-about-to-get-fryed-or-why-every-toot-is-also-a-potential-denial-of-service-attack/

Besides of that, the plugin does not handle following others yet, so you would only be able to post comments to direct replies to a post. So maybe your current plan makes the most sense (at least for now).

But if you have not enabled the plugin yet, we can not send a `Move`, because no one would receive it :)

So the question is more like: How can I setup the plugin so that it redirects users to a Mastodon Account before activating it? Do I understand that correctly?

**Comment by pkreissel** [comment]
> So the question is more like: How can I setup the plugin so that it redirects users to a Mastodon Account before activating it? Do I understand that correctly?

Yes we basically only need this part of the move protocol: 

> The user marks the new actor as accepting a move with the alsoKnownAs property.
> The user marks the old actor as having moved with the movedTo property

**Comment by pfefferle** [comment]
Ok, this might be even possible with a small custom plugin that adds the informations... But let me think about that a bit.

**Comment by pfefferle** [comment]
keep!

**Comment by pfefferle** [comment]
I think this is no "issue", so I will close it. We can continue the conversation anyways!
